LSPS5/README.md

```diff
@@ -0,0 +1,375 @@
+  and check that the current signature is not equal to
+  some remembered signature from a previous webhook
+  call.
+* MUST validate the signature against the message
+  template above, with the given timestamp and webhook
+  notification object.
+* MAY filter out particular notification methods by
+  its own policy.
+
+> **Rationale** The timestamp and signature checks are
+> intended to avoid replay attacks, where some third
+> party is able to copy the JSON object sent by the LSP
+> and replays it in order to frame the LSP for spamming
+> the client.
+
+The response to the webhook is unspecified.
+
+* For HTTPS webhooks (i.e. `POST`):
+  * The LSP MUST consider a `200 OK` response a
+    success, and MUST ignore the rest of the response,
+    including HTTP headers and the response body.
+  * If the response is not `200 OK`, the LSP SHOULD
+    log this as an unusual event, including the
+    HTTP headers and response body if any, and
+    MUST otherwise ignore the response.
+  * The LSP SHOULD NOT follow any redirection
+    responses.
+
+> **Rationale** As a notification, the user of the
+> client may ignore it, or disable notifications from
+> the client application entirely; thus, delivery of
+> the notification is never assured and it is pointless
+> to feed back anything to the LSP about whether the
+> notification was delivered to the notification
+> delivery service, the mobile OS server, or the client
+> mobile device.
+
+If the notification delivery service accepts the push
+notification, it MUST construct a suitable message to
+display to the user, and send that message, together
+with any required authentication and authorization, to
+the mobile OS server, for display as a push notification
+of the client.
+
+The client SHOULD process what is needed to
+react to the notification (e.g. connect to the LSP and
+accept the payment for a `lsps5.you_have_new_payments`
+notification) if the mobile environment allows it.
+
+#### Webhook Notification Methods
+
+The LSP MUST use one of the methods listed in this
+section for the webhook notification object.
+
+The LSP MUST NOT use any of these methods on the LSPS0
+interface; these notification methods are only for the
+webhook interface.
+
+Parameters are required unless otherwise stated.
+
+* `lsps5.you_have_just_registered` - The client has just
+  recently successfully called the `lsps5.set_webhook`
+  API.
+  No parameters `{}`.
+* `lsps5.you_have_new_payments` - The client has one or
+  more payments pending to be received.
+  Parameters:
+  * `total_msat` - The total amount, in millisatoshi,
+    that the client may receive.
+* `lsps5.your_contract_expires_soon` - There is an HTLC or
+  other time-bound contract, in either direction, on
+  one of the channels between the client and the LSP,
+  and it is within 24 blocks of being timed out, and
+  the timeout would cause a channel closure.
+  Parameters:
+  * `timeout` - The blockheight at which the LSP would
```

Comment:
See previous, same argument. If the client cannot make TCP/IP connection, it cannot fail the HTLC anyway. If it CAN make a TCP/IP connection, then it can get the information from the LSP directly instead of via the notification. So --- just how restricted is the client on a mobile device? Can it open TCP/IP? How much CPU time does it have? Can it do lookups on its local database? etc.

Reply:
ok.